vllm/config.py

```diff
@@ -321,6 +331,10 @@ def get_num_layers(self, parallel_config: "ParallelConfig") -> int:
         total_num_hidden_layers = self.hf_text_config.num_hidden_layers
         return total_num_hidden_layers // parallel_config.pipeline_parallel_size
 
+    def set_num_lookahead_tokens(self, num_lookahead_tokens: int):
```

Comment:
Can you describe the relationship between num_lookahead_tokens and num medusa heads?

For context, the `num_lookahead_tokens` value is used to allocate KV space for speculative tokens. Since Medusa does not use KV, we shouldn't require this to be equal to num heads.

Reply:
This is to allow the use of lower `num_heads` than in the medusa checkpoint. In case `num_lookahead_tokens` < `num_heads`, loading the additional heads into GPU memory would be inefficient. So we set the `num_heads` = `num_lookahead_tokens` and load only the heads that are required.